crates/devices/virtio-console/src/console.rs

```diff
@@ -0,0 +1,516 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Virtio console chain processing abstraction.
+//!
+//! This module offers an abstraction for working with the descriptor chains of the receive and
+//! transmit queues.
+//!  * The transmit queue is used to send data from the driver to the device and it's buffers are
+//!    read-only for the device.
+//!  * The receive queue is used to send data from the device to the driver and it's buffers are
+//!    write-only for the device.
+
+use std::cmp;
+use std::fmt::{self, Display};
+use std::io::{stdout, Stdout, Write};
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+use virtio_queue::DescriptorChain;
+use vm_memory::{Bytes, GuestMemory, GuestMemoryError};
+
+/// Console device errors.
+#[derive(Debug)]
+pub enum Error {
+    /// Failed to write slice to guest memory.
+    WriteToGuestFailed(GuestMemoryError),
+    /// Write to output sink failed.
+    WriteToOutputFailed(GuestMemoryError),
+    /// Transmitq descriptor chain length has overflown
+    ChainLengthOverflow,
+    /// Input buffer maximum capacity exceeded
+    BufferCapacityExceeded,
+    /// Read only descriptor that protocol says to write to.
+    UnexpectedReadOnlyDescriptor,
+    /// Write only descriptor that protocol says to read from.
+    UnexpectedWriteOnlyDescriptor,
+    /// The constructor's parameter for maximum capacity is 0
```

Comment:
I think we are returning this error also when the capacity > max one.

Reply:
We should update both the comment and the implementation of display below to correct the error message.